pkg/kubelet/cm/container_manager.go

```diff
@@ -52,6 +54,10 @@ type ContainerManager interface {
 
 	// GetNodeAllocatable returns the amount of compute resources that have to be reserved from scheduling.
 	GetNodeAllocatableReservation() v1.ResourceList
+
+	// UpdateQOSCgroups performs housekeeping updates to ensure that the top
+	// level QoS containers have their desired state in a thread-safe way
+	UpdateQOSCgroups() error
```

Comment:
i also want to go on the record that "container" is not defined in the kernel, but "cgroup" is.  and container is an overloaded term in a project like this.